miio/gateway.py

```diff
@@ -646,71 +646,112 @@ class GatewayLight(GatewayDevice):
+        rgb = int_to_rgb(state_int)
+        if brightness > 0:
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command()
+    def night_light_status(self):
+        """Get status of the night light."""
+        # The night light controls the same LEDs as the rgb light, but has a seprate memory for brightness and color.
+        # Changing the rgb light does not effect the stored state of the night light, while changing the night light does effect the state of the rgb light.
+        # Therefore this command only gives the correct status of the LEDs if the last command was a night light command and not a rgb light command, otherwise it gives the stored values of the night light.
+        # Returns {"is_on": false, "brightness": 0, "rgb": (0, 0, 0)} when night light is off
+        state_int = self._gateway.send("get_night_light_rgb").pop()
+        brightness = int_to_brightness(state_int)
+        rgb = int_to_rgb(state_int)
+        if brightness > 0:
+            is_on = True
+        else:
+            is_on = False
+        return {"is_on": is_on, "brightness": brightness, "rgb": rgb}
+
+    @command(
+        click.argument("brightness", type=int),
+        click.argument("rgb", type=(int, int, int)),
+    )
+    def set_rgb(self, brightness, rgb):
+        """Set gateway light using brightness and rgb tuple."""
+        brightness_and_color = brightness_and_color_to_int(brightness, rgb)
+        return self._gateway.send("set_rgb", [brightness_and_color])
+
+    @command(
+        click.argument("brightness", type=int),
+        click.argument("rgb", type=(int, int, int)),
+    )
+    def set_night_light(self, brightness, rgb):
+        """Set gateway night light using brightness and rgb tuple."""
+        brightness_and_color = brightness_and_color_to_int(brightness, rgb)
+        return self._gateway.send("set_night_light_rgb", [brightness_and_color])
+
+    @command(click.argument("brightness", type=int))
+    def set_rgb_brightness(self, brightness):
+        """Set gateway light brightness (0-100)."""
+        if 100 < brightness < 0:
+            raise Exception("Brightness must be between 0 and 100")
+        current_color = self.rgb_status()["rgb"]
+        return self.set_rgb(brightness, current_color)
+
+    @command(click.argument("brightness", type=int))
+    def set_night_light_brightness(self, brightness):
+        """Set night light brightness (0-100)."""
+        if 100 < brightness < 0:
+            raise Exception("Brightness must be between 0 and 100")
+        current_color = self.night_light_status()["rgb"]
+        return self.set_night_light(brightness, current_color)
 
     @command(click.argument("color_name", type=str))
-    def set_night_light_color(self, color_name):
-        """Set night light color using color name (red, green, etc)."""
+    def set_rgb_color(self, color_name):
```

Comment:
rgb consists of three separate components (r, g, and b), so shouldn't this rather be `set_color`?

Reply:
I wanted to be consistant between "rgb" and "night_light"
All functions begin with `set_rgb` or `set_night_light`

So I think this is the correct name

Reply:
Fair enough!